mapstruct/src/test/java/com/baeldung/expression/mapper/LicenseMapperUnitTest.java

```diff
@@ -0,0 +1,66 @@
+package com.baeldung.expression.mapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+import org.mapstruct.factory.Mappers;
+
+import com.baeldung.expression.dto.LicenseDto;
+import com.baeldung.expression.model.License;
+
+class LicenseMapperUnitTest {
+
+    LicenseMapper licenseMapper = Mappers.getMapper(LicenseMapper.class);
+
+    @Test
+    void givenLicenseDtoWithStartDateAndWithoutEndDate_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithDefaultEndDate() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .startDate(LocalDateTime.now())
+            .build());
+        assertThat(license).isNotNull();
+        assertThat(license.getEndDate()
+            .toLocalDate()).isEqualTo(LocalDate.now()
+            .plusYears(1));
+    }
+
+    @Test
+    void givenLicenseDtoWithEndDateAndWithoutStartDate_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithDefaultStartDate() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .endDate(LocalDateTime.now()
+                .plusYears(2))
+            .build());
+        assertThat(license).isNotNull();
+        assertThat(license.getStartDate()
+            .toLocalDate()).isEqualTo(LocalDate.now());
+    }
+
+    @Test
+    void givenLicenseDtoWithoutStartDateAndEndDate_WhenMapperMethodIsInvoked_ThenLicenseShouldBePopulatedWithDefaultDetails() {
+        License license = licenseMapper.toLicense(LicenseDto.builder()
+            .build());
+        assertThat(license).isNotNull()
+            .satisfies(l -> {
```

Comment:
As per the other tests, remove the satisfies and have the assertions line-by-line instead (unless this was left on purpose, happy to understand the reason if so)

Reply:
Besides this, looks good, ready to merge afterwards

Reply:
sure done, no specific reason but happy to change as per your comments